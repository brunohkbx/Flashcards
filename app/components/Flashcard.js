import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, CardContent } from 'react-native-paper';
import CardAction from '../components/Card/CardAction';
import RightAlignedActions from '../components/Card/RightAlignedActions';
import { DeleteIcon } from './Icons';
import { primaryColor } from '../config/theme';
import { View }  from 'react-native';

const Content = styled(CardContent)`
  padding-top: 0;
  padding-bottom: 8;
  margin-horizontal: 8;
`;

const Flashcard = ({ children, onFlashcardDeleted }) => {
  return (
    <Card>
      <Content>
        {children}
      </Content>
      <RightAlignedActions>
        <CardAction>
          <DeleteIcon
            size={24}
            color={primaryColor}
            onPress={onFlashcardDeleted}
          />
        </CardAction>
      </RightAlignedActions>
    </Card>
  );
};

Flashcard.propTypes = {
  children: PropTypes.node,
  onFlashcardDeleted: PropTypes.func.isRequired
};

export default Flashcard;
