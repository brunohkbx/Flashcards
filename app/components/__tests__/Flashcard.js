import React from 'react';
import { shallow } from 'enzyme';
import Flashcard from '../Flashcard';

describe('Flashcard', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      onFlashcardDeleted: jest.fn()
    }, propOverrides);

    const wrapper = shallow(<Flashcard {...defaultProps} />);

    const deleteFlashcard = () => wrapper.find('DeleteIcon').simulate('press');

    return { wrapper, deleteFlashcard };
  }

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  test('DeleteIcon onPress calls onFlashcardDeleted', () => {
    const mockOnFlashcardDeleted = jest.fn();
    const { deleteFlashcard } = setup({ onFlashcardDeleted: mockOnFlashcardDeleted });

    deleteFlashcard();

    expect(mockOnFlashcardDeleted).toHaveBeenCalled();
  });
});
