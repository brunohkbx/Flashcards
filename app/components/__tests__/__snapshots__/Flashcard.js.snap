// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flashcard renders properly 1`] = `
<withTheme(Card)>
  <Styled(CardContent) />
  <Styled(CardActions)>
    <Styled(Component)>
      <DeleteIcon
        color="#3f51b5"
        onPress={[MockFunction]}
        size={24}
      />
    </Styled(Component)>
  </Styled(CardActions)>
</withTheme(Card)>
`;
